src/Rings/MPolyMap/AffineAlgebras.jl

```diff
@@ -0,0 +1,220 @@
+
+function preimage(f::AffAlgHom, I::Union{MPolyIdeal, MPolyQuoIdeal})
+  @req base_ring(I) === codomain(f) "Parent mismatch"
+  D = domain(f)
+  salghom = _singular_algebra_morphism(f)
+  CS = codomain(salghom)
+  V = gens(I)
+  Ix = Singular.Ideal(CS, CS.(V))
+  prIx = Singular.preimage(salghom, Ix)
+  return ideal(D, D.(gens(prIx)))
+end
+
+################################################################################
+#
+#  Kernel
+#
+################################################################################
+
+function kernel(f::AffAlgHom)
+  get_attribute!(f, :kernel) do
+    C = codomain(f)
+    return preimage(f, ideal(C, [zero(C)]))
+  end # TODO: need some ideal_type(domain(f)) here :)
+end
+
+##############################################################################
+#
+#  Injectivity
+#
+##############################################################################
+
+@doc Markdown.doc"""
+    isinjective(F::AlgHom)
+
+Return `true` if `F` is injective, `false` otherwise.
+"""
+function isinjective(F::AffAlgHom)
+   G = gens(kernel(F))
+   for i in 1:length(G)
+      !iszero(G[i]) && return false
+   end
+   return true
+end
+
+# Helper function related to the computation of surjectivity, preimage etc.
+# Stores the necessary data in groebner_data, resp. groebner_data_lex
+function _groebner_data(F::AffAlgHom, ord::Symbol)
+   r = domain(F)
+   s = codomain(F)
+   n = ngens(r)
+   m = ngens(s)
+   return get_attribute!(F, ord) do
+     (S, I, W, _) = _ring_helper(s, zero(s), _images(F))
+     # Build auxilliary objects
+     (T, inc, J) = _containement_helper(S, n, m, I, W, ord)
+     D = normal_form([gen(T, i) for i in 1:m], J)
+     A = [zero(r) for i in 1:m]
+     B = [gen(r, i) for i in 1:n]
+     pr = hom(T, r, vcat(A, B))
+     groebner_data_lex = (T, inc, pr, J, D)
+     return groebner_data_lex
+   end
+ end
+
+################################################################################
+#
+#  Surjectivity
+#
+################################################################################
+
+function issurjective(F::AffAlgHom)
+   # Compute data necessary for computation
+   r = domain(F)
+   s = codomain(F)
+   n = ngens(r)
+   m = ngens(s)
+   (T, _, _, _, D) = _groebner_data(F, :degrevlex)
+
+   # Check if map is surjective
+
+   for i in 1:m
+      if !(leading_monomial(D[i]) < gen(T, m))
+         return false
+      end
+   end
+   return true
+end
+
+################################################################################
+#
+#  Bijectivity
+#
+################################################################################
+
+function isbijective(F::AffAlgHom)
+  return isinjective(F) && issurjective(F)
+end
+
+################################################################################
+#
+#  Finiteness
+#
+################################################################################
+
+@doc Markdown.doc"""
+    isfinite(F::AlgHom)
+
+Return `true` if `F` is finite, `false` otherwise.
```

Comment:
I realize this is a prototype, but I am just noting down all I see. Of course feel free to ignore whatever doesn't fit in your current time schedule, this can also be addressed later... that said:

I don't think it is common knowledge what a "finite algebra homomorphism" is (or at least I missed that class), so perhaps this could be explained here?